foreground.py

```diff
@@ -103,17 +104,17 @@ def __call__(self, frame):
                 std = np.std(frame)
                 foreground[frame < mn + std] = 0
 
-            self.frames_history = np.append(self.frames_history,np.expand_dims(frame, axis=0), axis=0)
+            self.frames_history.append(frame)
 
-            if self.frames_history.shape[0]> self.num_frames:
+            if len(self.frames_history) > self.num_frames:
                 self.window_sum -= self.frames_history[0]
-                self.frames_history = self.frames_history[1:]
+                self.frames_history.popleft()
 
             self.window_sum += self.frames_history[-1]
         return foreground
 
     def reset(self):
-        self.frames_history = None
+        self.frames_history = deque()
```

Comment:
Please notice that after the modification above the reset function should assign a new deque to frames_history instead of None.